network/message.go

```diff
@@ -14,8 +16,8 @@ type PluginContext struct {
 }
 
 // Reply sends back a message to an incoming message's incoming stream.
-func (ctx *PluginContext) Reply(message proto.Message) error {
-	return ctx.client.Reply(ctx.nonce, message)
+func (ctx *PluginContext) Reply(gCtx context.Context, message proto.Message) error {
+	return ctx.client.Reply(gCtx, ctx.nonce, message)
```

Comment:
This is incredibly confusing now, we have a noise context and the go context, and they're both `ctx`. Should probably rethink whether we want both to exist.

Reply:
I suggest we rename PluginContext to something else then. What do you think

Reply:
Better rename receiver `ctx` to `pctx` and rename `gCtx` to `ctx`.